Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs

```diff
@@ -0,0 +1,169 @@
+        _mockMapper = new Mock<IMapper>();
+
+        _useCase = new UpdateConnectedEntityUseCase(
+            _mockConnectedEntityRepository.Object,
+            _mockOrganisationRepository.Object,
+            _mockMapper.Object);
+    }
+
+    [Fact]
+    public async Task Execute_ThrowsUnknownOrganisationException_WhenOrganisationDoesNotExist()
+    {
+        var command = (organisationId: Guid.NewGuid(), connectedEntityId: Guid.NewGuid(), updateConnectedEntity: new UpdateConnectedEntity
+        {
+            EntityType = (ConnectedEntityType)0
+        });
+        _mockOrganisationRepository.Setup(repo => repo.Find(command.organisationId)).ReturnsAsync((Persistence.Organisation)null);
+
+        await Assert.ThrowsAsync<UnknownOrganisationException>(() => _useCase.Execute(command));
+    }
+
+    private static Persistence.Organisation FakeOrganisation(bool? withSupplierInfo = true)
+    {
+        Persistence.Organisation org = new()
+        {
+            Guid = Guid.NewGuid(),
+            Name = "FakeOrg",
+            Tenant = new Persistence.Tenant()
+            {
+                Guid = Guid.NewGuid(),
+                Name = "Tenant 101"
+            },
+            ContactPoints = [new Persistence.Organisation.ContactPoint { Email = "contact@test.org" }]
+        };
+
+        if (withSupplierInfo == true)
+        {
+            org.SupplierInfo = new Persistence.Organisation.SupplierInformation { CompletedRegAddress = true };
+        }
+
+        return org;
+    }
+
+    private static Persistence.ConnectedEntity FakeConnectedEntity(Persistence.Organisation organisation)
+    {
+        Persistence.ConnectedEntity connectedEntity = new()
+        {
+            Guid = Guid.NewGuid(),
+            EntityType = Persistence.ConnectedEntity.ConnectedEntityType.Individual,
+            SupplierOrganisation = organisation,
+            Addresses = new List<Persistence.ConnectedEntity.ConnectedEntityAddress>(),
+            Id = 1,
+            HasCompnayHouseNumber = true,
+            CreatedOn = DateTimeOffset.UtcNow,
+            UpdatedOn = DateTimeOffset.UtcNow,
+        };
+
+        return connectedEntity;
+    }
+
+    [Fact]
+    public async Task Execute_ThrowsUnknownConnectedEntityException_WhenConnectedEntityDoesNotExist()
+    {
+        var organisationId = Guid.NewGuid();
+        var connectedEntityId = Guid.NewGuid();
+        var updateConnectedEntity = new UpdateConnectedEntity
+        {
+            EntityType = (ConnectedEntityType)0
+        };
+        var command = (organisationId, connectedEntityId, updateConnectedEntity);
+
+        var organisation = FakeOrganisation();
+
+        _mockOrganisationRepository.Setup(repo => repo.Find(organisationId)).ReturnsAsync(organisation);
+        _mockConnectedEntityRepository.Setup(repo => repo.Find(organisationId, connectedEntityId)).ReturnsAsync((Persistence.ConnectedEntity)null);
```

Comment:
Static analysis reports a problem here: Converting null literal or possible null value to non-nullable type..